src/DecayTreeVectorIntegral.cxx

```diff
@@ -150,4 +165,30 @@ const RealIntegralElement integral(const DecayTreeVectorIntegral& dtvi)
     return I;
 }
 
+//-------------------------
+const RealIntegralElement integral(const DecayTreeVectorIntegral& dtvi, const DecayTreeVector& dtv)
+{
+    auto dts = dtvi.decayTrees();
```

Comment:
There's no need for the variable `dts` (which can just be replaced below with `dtvi.decayTrees()`.
But if you use it for shorter lines below, then certainly you should not create a new `DecayTreeVector` but instead call `const auto& dts = dtvi.decayTrees();`